src/otaclient/grpc/api_v2/ecu_status.py

```diff
@@ -364,19 +372,18 @@ def get_polling_waiter(self):
             or self.active_ota_update_present is set, return when one of the
             condition is met.
         """
+        _inner_wait_interval = 1
 
         async def _waiter():
-            if self.active_ota_update_present.is_set():
+            ecu_status_flags = self.ecu_status_flags
+            if ecu_status_flags.any_in_update.is_set():
                 await asyncio.sleep(ACTIVE_POLLING_INTERVAL)
                 return
 
-            try:
-                await asyncio.wait_for(
-                    self.active_ota_update_present.wait(),
-                    timeout=IDLE_POLLING_INTERVAL,
-                )
-            except asyncio.TimeoutError:
-                return
+            for _ in range(math.ceil(IDLE_POLLING_INTERVAL / _inner_wait_interval)):
+                if ecu_status_flags.any_in_update.is_set():
+                    return
+                await asyncio.sleep(_inner_wait_interval)
```

Comment:
Oh please take a look at the method clearly, the `return` statement is within the closure `_waiter`.
The outer layer of  `return _waiter` is just returning the closure.

Here I implement a polling_waiter factory(the `get_polling_waiter` method), and it will generate and return a async function for the caller to wait for certain time, according to the current multiple ECU status flags.

Reply:
Thank you, I misunderstood :)
In both before/after, `_waiter()` just return when timeout is occurred.